### PR TITLE
Build get_dataset_worker function and test

### PR DESCRIPTION
**Pull request by LauraZM9**
Ticket: [[Epic 1] Build function to call raw data file](https://trello.com/c/RwXjXktb)

This PR: 

- Build function to read worker data in parquet format (only two features to start with)
- Build test for the two features
- Generate a dummy dataset for testing

Next step: generate data for all 68 features and adapt function and test

### CODE REVIEW

File: tests/unit/test_prepare_workers.py [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
+import shutil
 from pyspark.sql import SparkSession
 from jobs import prepare_workers
 
+from tests.test_file_generator import generate_ascwds_worker_file
+
 
 class PrepareWorkersTests(unittest.TestCase):
+
+    TEST_ASCWDS_WORKER_FILE = "tests/test_data/domain=ascwdsdataset=worker"
```
Comment (elcorbs) [inline tests/unit/test_prepare_workers.py:11]: Is this missing a forward slash? `domain=ascwds/dataset=worker`

**Review by elcorbs — APPROVED** [review]
A couple of small comments on the tests but the functionality looks good.

File: tests/unit/test_prepare_workers.py [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
+    @classmethod
+    def tearDownClass(self):
+        try:
+            shutil.rmtree(self.TEST_ASCWDS_WORKER_FILE)
+        except OSError():
+            pass  # Ignore dir does not exist
 
     def test_main(self):
         result = prepare_workers.main("source", "destination")
         self.assertTrue(result)
```
Comment (garypolzin) [inline tests/unit/test_prepare_workers.py:26]: probably good to test the number of rows in the final output

**Review by garypolzin — APPROVED** [review]
looks good - I'm not 100% on my setup comment, it's what I used on another script when pairing with Adam

File: jobs/prepare_workers.py [hunk 3]
```diff
@@ -1,11 +1,29 @@
 import argparse
 import sys
 
+from utils import utils
+
+from pyspark.sql.functions import col
```
Comment (JimmySunDance) [inline jobs/prepare_workers.py:6]: Please move this above the project imports

File: jobs/prepare_workers.py [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
 
 def main(source, destination):
     return True
 
 
+def get_dataset_worker(worker_source):
```
Comment (JimmySunDance) [inline jobs/prepare_workers.py:12]: Do we want this to be `worker_source`

File: tests/unit/test_prepare_workers.py [hunk 5]
```diff
@@ -1,18 +1,30 @@
 import unittest
+import shutil
 from pyspark.sql import SparkSession
 from jobs import prepare_workers
 
+from tests.test_file_generator import generate_ascwds_worker_file
+
```
Comment (JimmySunDance) [inline tests/unit/test_prepare_workers.py:7]: Please organise imports as shown in `test_prepare_location.py`.

Comment (JimmySunDance) [inline tests/unit/test_prepare_workers.py:7]: Please organise imports as shown in `test_prepare_location.py`.

File: tests/test_file_generator.py [hunk 6]
```diff
@@ -374,3 +373,19 @@ def generate_ascwds_workplace_file(output_destination):
         df.coalesce(1).write.mode("overwrite").parquet(output_destination)
 
     return df
+
+
+def generate_ascwds_worker_file(output_destination):
```
Comment (JimmySunDance) [inline tests/test_file_generator.py:378]: There is already a `generate_worker_parquet()` function in this folder. Is it worth combining with/utilising this func instead?

Comment (JimmySunDance) [inline tests/test_file_generator.py:378]: There is already a `generate_worker_parquet()` function in this folder. Is it worth combining with/utilising this func instead?

**Review by JimmySunDance — COMMENTED** [review]
Looks good - a few minor comments

**Pull request merged**